tests/src/main/java/net/neoforged/neoforge/debug/attachment/AttachmentTests.java

```diff
@@ -319,4 +319,21 @@ static void itemAttachmentPresence(final DynamicTest test, final RegistrationHel
             helper.succeed();
         });
     }
+
+    @GameTest
+    @EmptyTemplate
+    @TestHolder(description = "Ensures that custom cloning behaviour works")
+    static void itemAttachmentCloning(final DynamicTest test, final RegistrationHelper reg) {
+        var attachmentType = reg.registrar(NeoForgeRegistries.Keys.ATTACHMENT_TYPES)
+                .register("test_int", () -> AttachmentType.builder(() -> 0).cloner((holder, i) -> 2 * i).build());
```

Comment:
This is fine for test purposes, but maybe add a comment saying that mods shouldn't do this as it violates the contract of `IAttachmentCloner#clone`.

Reply:
Currently the IAttachmentCloner doesn't require the AttachmentType to have a serializer, so that would have to be changed too, to make this comment valid. I do agree that a comment saying, that modifying the attachment on cloning isn't expected behavior and is only used in this test for easy detectability, is a good thing.